src/features/posts/commentsSlice.ts

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+import * as commentsApi from '../../api/comments';
+
+export type CommentsState = {
+  comments: Comment[];
+  status: 'idle' | 'loading' | 'failed';
+};
+
+const initState: CommentsState = {
+  comments: [],
+  status: 'idle',
+};
+
+export const getCommentsAsync = createAsyncThunk(
+  'comments/getComments',
+  async (postId: number) => {
+    const comments = await commentsApi.getPostComments(postId);
+
+    return comments;
+  },
+);
+
+export const addCommentAsync = createAsyncThunk(
+  'comments/addComment',
+  async (
+    {
+      name, email, body, postId,
+    }: { name: string; email: string; body: string; postId: number },
+  ) => {
+    const response = await commentsApi.createComment({
+      name,
+      email,
+      body,
+      postId,
+    });
+
+    return response;
+  },
+);
+
+export const deleteCommentAsync = createAsyncThunk(
+  'comments/deleteComment',
+  async (commentId: number) => {
+    await commentsApi.deleteComment(commentId);
+
+    return commentId;
+  },
+);
+
+export const commentsSlice = createSlice({
+  name: 'comments',
+  initialState: initState,
+  reducers: {
+    addComment: (state, action: PayloadAction<Comment>) => {
+      state.comments.push(action.payload);
+    },
+
+    removeComment: (state, action: PayloadAction<number>) => {
+      state.comments = [...state.comments]
```

Comment:
```suggestion
      state.comments = state.comments
```
redundant copy, the filter returns the new array